src/drivers/driver_zephyr.c

```diff
@@ -5,64 +5,36 @@
  * This software may be distributed under the terms of the BSD license.
  * See README for more details.
  */
+#include <zephyr/kernel.h>
+
 #include "includes.h"
 #include "utils/common.h"
 #include "eloop.h"
 #include "driver_zephyr.h"
+#include "supp_main.h"
 
-K_MBOX_DEFINE(wpa_supp_mbox);
 #define SCAN_TIMEOUT 30
 
-static void wpa_supp_drv_mbox_msg_handler(void *eloop_ctx,
-					  void *timeout_ctx)
-{
-	struct k_mbox_msg msg;
-	struct zep_wpa_supp_mbox_msg_data mbox_msg_data;
-
-	/* Prepare to receive message */
-	msg.size = sizeof(mbox_msg_data);
-	msg.rx_source_thread = K_ANY;
-
-	k_mbox_get(&wpa_supp_mbox,
-		   &msg,
-		   &mbox_msg_data,
-		   K_FOREVER);
-
-	mbox_msg_data.cb(mbox_msg_data.ctx,
-			 mbox_msg_data.data,
-			 0);
-}
-
-
-void wpa_supp_event_handler(void *ctx,
-			    void *data,
-			    void *cb)
+void wpa_supplicant_event_wrapper(void *ctx,
+				enum wpa_event_type event,
+				union wpa_event_data *data)
 {
-	struct k_mbox_msg send_msg;
-	struct zep_wpa_supp_mbox_msg_data mbox_msg_data;
-
-	mbox_msg_data.ctx = ctx;
-	mbox_msg_data.data = data;
-	mbox_msg_data.cb = cb;
-
-	/* Prepare to send message */
-	send_msg.size = sizeof(mbox_msg_data);
-	send_msg.tx_data = &mbox_msg_data;
-	send_msg.tx_block.data = NULL;
-	send_msg.tx_target_thread = K_ANY;
-
-	eloop_register_timeout(0,
-			       500,
-			       wpa_supp_drv_mbox_msg_handler,
-			       NULL,
-			       NULL);
-
-	k_mbox_put(&wpa_supp_mbox,
-		   &send_msg,
-		   K_FOREVER);
+	struct wpa_supplicant_event_msg msg = { 0 };
+
```

Comment:
Might be better to add a defensive check to see that ctx is non-NULL.

Reply:
`ctx` cannot be NULL if we are in the `driver` layer.